### ISSUE TITLE
Autoformat of Nuitka source code gives TypeError and UnicodeDecodeError

### ISSUE THREAD

**Issue by bksahu** [Nuitka/Nuitka#334]
I'm trying to autoformat my code. On executing without any flags I get `TypeError`
```
$ ./bin/autoformat-nuitka-source 
Working on: ['bin', 'nuitka']
Consider bin/autoformat-nuitka-source: Traceback (most recent call last):
  File "./bin/autoformat-nuitka-source", line 34, in <module>
    main()
  File "./nuitka/tools/quality/autoformat/__main__.py", line 100, in main
    autoformat(filename, git_stage=False, abort=options.abort)
  File "./nuitka/tools/quality/autoformat/Autoformat.py", line 332, in autoformat
    output_file.write(old_code)
TypeError: a bytes-like object is required, not 'str'

```

I have also tried to autoformat more than one dir, I get `UnicodeDecodeError`.
```
$ ./bin/autoformat-nuitka-source nuitka testsWorking on: ['nuitka', 'tests']
Traceback (most recent call last):
  File "./bin/autoformat-nuitka-source", line 34, in <module>
    main()
  File "./nuitka/tools/quality/autoformat/__main__.py", line 94, in main
    scanTargets(positional_args, (".py", ".scons", ".rst", ".txt"))
  File "./nuitka/tools/quality/ScanSources.py", line 74, in scanTargets
    for value in addFromDirectory(positional_arg, suffixes, blacklist):
  File "./nuitka/tools/quality/ScanSources.py", line 61, in addFromDirectory
    shebang = getShebangFromFile(fullpath)
  File "./nuitka/utils/Shebang.py", line 70, in getShebangFromFile
    return getShebangFromSource(f.readline())
  File "/home/bksahu/anaconda3/envs/nuitka/lib/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa2 in position 41: invalid start byte

```

**python -m nuitka --version**
0.6.4rc1
Python: 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
Executable: /home/bksahu/anaconda3/envs/nuitka/bin/python
OS: Linux
Arch: x86_64

**Update**
After rebasing I'm getting `TypeError: write() argument must be str, not bytes`

**Comment by kayhayen** [comment]
So I fixed this on the factory branch, and will push to develop soon. Rebase your PRs to that, and then you ought to be able to follow the instructions to install the commit hook.

I recently changed the autoformat so it is usable from the pre-commit hook and only modifies the staged content in that case, no more git add. Re-run that install script for the hook. If it ever causes trouble, as a workaround `rm .git/hooks/pre-commit` will be there.

I didn't make a part of the `CONTRIBUTING.md` yet, as it is very much in flux. Getting anything to work with Python2/Python3 on Linux and Windows, is rather close to rocket science, and I am not a rocket scientist. But I feel it might be there really soon now.

I think one missing bit might that it looses "x" bits, at least I have seen that happen to me once. But when used with care, it's a miracle tool. Esp. getting commit hooks to work on Windows, is woah, worth its own whining article. :)
